### PR TITLE
convert case_id fields to foreign keys

### PR DESCRIPTION
**Pull request by snopoke**
@calellowitz 

https://github.com/dimagi/commcare-hq/pull/11911/files/e47e30182d18c444e8e9cb73129c4c41170ac21c..7ddf5ec5af25e84382b8b2ada4770ecb30e365e6#r65846802

cc @czue

### CODE REVIEW

File: corehq/form_processor/migrations/0060_convert_case_ids_to_foreign_keys.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        ('form_processor', '0059_remove_ledgervalue_location_id'),
+    ]
+
+    operations = [
+        # The other way to convert a field to a foreign key involves creating a new column
+        # doing a data migration and then dropping the old column which seemed ridiculous to me
+        HqRunSQL(
+            'ALTER TABLE "form_processor_ledgervalue" '
+            'ADD CONSTRAINT "cd40c15ceaad5d793e09d0b69eb4ed88" FOREIGN KEY ("case_id") '
+            'REFERENCES "form_processor_commcarecasesql" ("case_id") DEFERRABLE INITIALLY DEFERRED',
```
Comment (snopoke) [inline corehq/form_processor/migrations/0060_convert_case_ids_to_foreign_keys.py]: I got this SQL by doing a running `sqlmigrate` command with the state operations below. The name of the constraints are consistent across environments.

**Comment by calellowitz** [comment]
test failures look real

**Comment by czue** [comment]
lgtm

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**